interface/forms/fee_sheet/new.php

```diff
@@ -879,21 +619,90 @@ function setJustify(seljust) {
  theopts[j++] = new Option('Clear','',false,false);
 }
 
+// Determine if there are any charges in this visit.
+function hasCharges() {
+ var f = document.forms[0];
+ for (var i = 0; i < f.elements.length; ++i) {
+  var elem = f.elements[i];
+  if (elem.name.indexOf('[price]') > 0) {
+   var fee = Number(elem.value);
+   if (!isNaN(fee) && fee != 0) return true;
+  }
+ }
+ return false;
+}
+
+// Function to check if there are any charges in the form, and to enable
+// or disable the Save and Close button accordingly.
+//
+function setSaveAndClose() {
+ var f = document.forms[0];
+ if (!f.bn_save_close) return;
+ if (hasCharges()) {
+  f.form_has_charges.value = '1';
+  f.bn_save_close.value = '<?php echo xla('Save and Checkout'); ?>';
+ }
+ else {
+  f.form_has_charges.value = '0';
+  f.bn_save_close.value = '<?php echo xla('Save and Close'); ?>';
+ }
+}
+
+// Open the add-event dialog.
+function newEvt() {
+ var f = document.forms[0];
+ var url = '../../main/calendar/add_edit_event.php?patientid=<?php echo urlencode($fs->pid); ?>';
```

Comment:
Hmm, looks like github stripped out my scripting attack. :)
What was in the last urlencode was:
[script]alert('gotcha');[/script]
but with angle brackets instead of square brackets.